docs/conf.py

```diff
@@ -163,6 +163,8 @@ def swap_prefix(file, old, new):
 def setup(app):
     import os
 
+    os.popen("git submodule update --init --recursive")
```

Comment:
ouch! This will run asynchronously? I think the concern of setting up the source folder should be kept out of the Sphinx configuration file. Do it before running sphinx.